### PR TITLE
Add File API

### PR DESCRIPTION
**Pull request by OliverUrsell**
Add support for the file api, including a function to upload a file to HubSpot and retrieve its link.

### CODE REVIEW

File: file_test.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+					DuplicateValidationScope:    "EXACT_FOLDER",
+				}
+
+				var gotOptions FileUploadOptions
+				err := json.Unmarshal([]byte(req.FormValue("options")), &gotOptions)
+				if err != nil {
+					t.Errorf("Error unmarshalling request options to a FileUploadOptions struct: %s" + err.Error())
+				}
+
+				if !reflect.DeepEqual(expectedOptions, gotOptions) {
```
Comment (d-lowl) [inline file_test.go:54]: Should avoid reflection, and use `cmp.Equal` instead

File: file_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+				if err != nil {
+					t.Errorf("Error getting file from form request: %s", err.Error())
+				}
+
+				buf := bytes.NewBuffer(nil)
+				if _, err := io.Copy(buf, file); err != nil {
+					return nil, err
+				}
+
+				if !reflect.DeepEqual(buf.Bytes(), fileContent) {
```
Comment (d-lowl) [inline file_test.go:69]: Same here, avoid reflection

**Review by d-lowl — CHANGES_REQUESTED** [review]
Comments regarding the use of reflection, otherwise good to go

File: file_test.go (hunk 1 above)
Comment (OliverUrsell) [reply file_test.go]: Updated the use of reflection to use cmp

File: file_test.go (hunk 2 above)
Comment (OliverUrsell) [reply file_test.go]: See https://github.com/fuzzylabs/go-hubspot/pull/2#discussion_r747502085

**Review by d-lowl — APPROVED** [review]

**Review by archena — APPROVED** [review]

**Pull request merged**